jetbrains.jdk: add updateScript and a test

liff opened the pull request:
The update script uses the `jbr/release` file in IDEA Ultimate’s current release tarball to resolve the latest JBR version JetBrains is currently using in their IDEs.

This only adds the update script, the package version remains the same.

I also added a NixOS test that attempts to verify that JCEF works.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

karolchmist commented:
Result of `nixpkgs-review pr 301653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

karolchmist commented:
Looks good! But is there a way to test it manually? I use idea, can I somehow test it with it?

liff commented:
This does not change the built package in any way, but if you want to test the changes, you can try the following:

1. Check out this PR
    ```shell
    $ nix-shell -p gh --command 'gh pr checkout 302592'
    ```
2. [Test the update script](https://nixos.org/manual/nixpkgs/stable/#var-passthru-updateScript-execution):
    ```shell
    $ nix-shell maintainers/scripts/update.nix --argstr package jetbrains.jdk
    ```
3. Build the updated runtime package.
    ```shell
    $ nix-build -A jetbrains.jdk
    ```
4. [Run the NixOS tests](https://nixos.org/manual/nixos/stable/#sec-running-nixos-tests)
    ```shell
    $ nix-build -A nixosTests.jetbrains-jdk
    ```

Building the JDK takes a while, so skip 2 and 3 if you don’t want to test that 😄

karolchmist commented:
Running `nix-build -A nixosTests.jetbrains-jdk` results in no errors :+1:

liff committed:
jetbrains.jdk: add updateScript and a test

The update script uses the `jbr/release` file in IDEA Ultimate’s
current release tarball to resolve the latest JBR version JetBrains
is currently using in their IDEs.

liff committed:
jetbrains.jdk: add a NixOS test

The test attempts to verify that JCEF works by building a simple browser
application and checking that it can display the `chrome://system` page.

liff commented:
I'll return to this later.
